test/test_videoapi.py

```diff
@@ -0,0 +1,213 @@
+    """
+
+    file_name = name.split("/")[1]
+    download_url("http://fate.ffmpeg.org/fate-suite/" + name, path, file_name)
+    return os.path.join(path, file_name)
+
+
+test_videos = {
+    "RATRACE_wave_f_nm_np1_fr_goo_37.avi": GroundTruth(
+        duration=2.0, video_fps=30.0, audio_sample_rate=None
+    ),
+    "SchoolRulesHowTheyHelpUs_wave_f_nm_np1_ba_med_0.avi": GroundTruth(
+        duration=2.0, video_fps=30.0, audio_sample_rate=None
+    ),
+    "TrumanShow_wave_f_nm_np1_fr_med_26.avi": GroundTruth(
+        duration=2.0, video_fps=30.0, audio_sample_rate=None
+    ),
+    "v_SoccerJuggling_g23_c01.avi": GroundTruth(
+        duration=8.0, video_fps=29.97, audio_sample_rate=None
+    ),
+    "v_SoccerJuggling_g24_c01.avi": GroundTruth(
+        duration=8.0, video_fps=29.97, audio_sample_rate=None
+    ),
+    "R6llTwEh07w.mp4": GroundTruth(
+        duration=10.0, video_fps=30.0, audio_sample_rate=44100
+    ),
+    "SOX5yA1l24A.mp4": GroundTruth(
+        duration=11.0, video_fps=29.97, audio_sample_rate=48000
+    ),
+    "WUzgd7C1pWA.mp4": GroundTruth(
+        duration=11.0, video_fps=29.97, audio_sample_rate=48000
+    ),
+}
+
+
+@unittest.skipIf(_HAS_VIDEO_OPT is False, "Didn't compile with ffmpeg")
+class TestVideoApi(unittest.TestCase):
+    def test_predefined_metadata(self):
+        """
+        Test that the source metadata corresponds to the one returned
+        by the new video decoder API.
+        """
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+            reader = VideoReader(full_path, "video")
+            reader_md = reader.get_metadata()
+            self.assertAlmostEqual(
+                config.video_fps, reader_md["video"]["fps"][0], delta=0.0001
+            )
+            self.assertAlmostEqual(
+                config.duration, reader_md["video"]["duration"][0], delta=0.5
+            )
+
+    def test_frame_reading(self):
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+
+            av_reader = av.open(full_path)
+
+            if av_reader.streams.video:
+                video_reader = VideoReader(full_path, "video")
+                for av_frame in av_reader.decode(av_reader.streams.video[0]):
+                    vr_frame = next(video_reader)
+
+                    self.assertAlmostEqual(
+                        float(av_frame.pts * av_frame.time_base),
+                        vr_frame["pts"],
+                        delta=0.1,
+                    )
+
+                    av_array = torch.tensor(av_frame.to_rgb().to_ndarray()).permute(
+                        2, 0, 1
+                    )
+                    vr_array = vr_frame["data"]
+                    mean_delta = torch.mean(
+                        torch.abs(av_array.float() - vr_array.float())
+                    )
```

Comment:
In general I prefer to use `torch.max` to measure the difference between two tensors.
But in here, given that your mean difference is `2.5`, this might mean that we might have very large differences between the tensor values, is that correct?

Reply:
since the range is 0-255, this ensures that the difference between the decoded frames is less than 1% on avg. 

Using torch.max might make issue if the default decoding for one decoder is 0 and the other 255 (borders for example)